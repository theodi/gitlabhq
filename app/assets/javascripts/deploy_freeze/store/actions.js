import * as types from './mutation_types';
import Api from '~/api';
import { deprecatedCreateFlash as createFlash } from '~/flash';
import { __ } from '~/locale';

export const requestAddFreezePeriod = ({ commit }) => {
  commit(types.REQUEST_ADD_FREEZE_PERIOD);
};

export const receiveAddFreezePeriodSuccess = ({ commit }) => {
  commit(types.RECEIVE_ADD_FREEZE_PERIOD_SUCCESS);
};

export const receiveAddFreezePeriodError = ({ commit }, error) => {
  commit(types.RECEIVE_ADD_FREEZE_PERIOD_ERROR, error);
};

export const addFreezePeriod = ({ state, dispatch, commit }) => {
  dispatch('requestAddFreezePeriod');

  return Api.createFreezePeriod(state.projectId, {
    freeze_start: state.freezeStartCron,
    freeze_end: state.freezeEndCron,
    cron_timezone: state.selectedTimezoneIdentifier,
  })
    .then(() => {
      dispatch('receiveAddFreezePeriodSuccess');
      commit(types.RESET_MODAL);
      dispatch('fetchFreezePeriods');
    })
    .catch(error => {
      createFlash(__('Error: Unable to create deploy freeze'));
      dispatch('receiveAddFreezePeriodError', error);
    });
};

export const fetchFreezePeriods = ({ commit, state }) => {
  commit(types.REQUEST_FREEZE_PERIODS);

  return Api.freezePeriods(state.projectId)
    .then(({ data }) => {
      commit(types.RECEIVE_FREEZE_PERIODS_SUCCESS, data);
    })
    .catch(() => {
      createFlash(__('There was an error fetching the deploy freezes.'));
    });
};

export const setSelectedTimezone = ({ commit }, timezone) => {
  commit(types.SET_SELECTED_TIMEZONE, timezone);
};

export const setFreezeStartCron = ({ commit }, { freezeStartCron }) => {
  commit(types.SET_FREEZE_START_CRON, freezeStartCron);
};

export const setFreezeEndCron = ({ commit }, { freezeEndCron }) => {
  commit(types.SET_FREEZE_END_CRON, freezeEndCron);
};

export const resetModal = ({ commit }) => {
  commit(types.RESET_MODAL);
};
