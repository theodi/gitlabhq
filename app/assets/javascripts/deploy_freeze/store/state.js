export default ({
  projectId,
  freezePeriods = [],
  timezoneData = [],
  selectedTimezone = '',
  selectedTimezoneIdentifier = '',
  freezeStartCron = '',
  freezeEndCron = '',
}) => ({
  projectId,
  freezePeriods,
  timezoneData,
  selectedTimezone,
  selectedTimezoneIdentifier,
  freezeStartCron,
  freezeEndCron,
});
