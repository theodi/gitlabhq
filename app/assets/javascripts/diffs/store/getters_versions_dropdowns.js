import { __, n__, sprintf } from '~/locale';
import { getParameterByName, parseBoolean } from '~/lib/utils/common_utils';
import { DIFF_COMPARE_BASE_VERSION_INDEX, DIFF_COMPARE_HEAD_VERSION_INDEX } from '../constants';

export const selectedTargetIndex = state =>
  state.startVersion?.version_index || DIFF_COMPARE_BASE_VERSION_INDEX;

export const selectedSourceIndex = state => state.mergeRequestDiff.version_index;

export const diffCompareDropdownTargetVersions = (state, getters) => {
  // startVersion only exists if the user has selected a version other
  // than "base" so if startVersion is null then base must be selected

  const defaultMergeRefForDiffs = window.gon?.features?.defaultMergeRefForDiffs || false;
  const diffHeadParam = getParameterByName('diff_head');
  const diffHead = parseBoolean(diffHeadParam) || (!diffHeadParam && defaultMergeRefForDiffs);
  const isBaseSelected = !state.startVersion && !diffHead;
  const isHeadSelected = !state.startVersion && diffHead;
  let baseVersion = null;

  if (
    !defaultMergeRefForDiffs ||
    (defaultMergeRefForDiffs && !state.mergeRequestDiff.head_version_path)
  ) {
    baseVersion = {
      versionName: state.targetBranchName,
      version_index: DIFF_COMPARE_BASE_VERSION_INDEX,
      href: state.mergeRequestDiff.base_version_path,
      isBase: true,
      selected:
        isBaseSelected || (defaultMergeRefForDiffs && !state.mergeRequestDiff.head_version_path),
    };
  }

  const headVersion = {
    versionName: state.targetBranchName,
    version_index: DIFF_COMPARE_HEAD_VERSION_INDEX,
    href: state.mergeRequestDiff.head_version_path,
    isHead: true,
    selected: isHeadSelected,
  };
  // Appended properties here are to make the compare_dropdown_layout easier to reason about
  const formatVersion = v => {
    return {
      href: v.compare_path,
      versionName: sprintf(__(`version %{versionIndex}`), { versionIndex: v.version_index }),
      selected: v.version_index === getters.selectedTargetIndex,
      ...v,
    };
  };

  return [
    ...state.mergeRequestDiffs.slice(1).map(formatVersion),
    baseVersion,
    state.mergeRequestDiff.head_version_path && headVersion,
  ].filter(a => a);
};

export const diffCompareDropdownSourceVersions = (state, getters) => {
  // Appended properties here are to make the compare_dropdown_layout easier to reason about
  return state.mergeRequestDiffs.map((v, i) => ({
    ...v,
    href: v.version_path,
    commitsText: n__(`%d commit,`, `%d commits,`, v.commits_count),
    versionName:
      i === 0
        ? __('latest version')
        : sprintf(__(`version %{versionIndex}`), { versionIndex: v.version_index }),
    selected: v.version_index === getters.selectedSourceIndex,
  }));
};
