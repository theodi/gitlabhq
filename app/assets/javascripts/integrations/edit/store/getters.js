export const isInheriting = state => (state.defaultState === null ? false : !state.override);

export const isSavingOrTesting = state => state.isSaving || state.isTesting;

export const propsSource = (state, getters) =>
  getters.isInheriting ? state.defaultState : state.customState;

export const currentKey = (state, getters) => (getters.isInheriting ? 'admin' : 'custom');
