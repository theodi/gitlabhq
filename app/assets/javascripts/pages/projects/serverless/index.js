import ServerlessBundle from '~/serverless/serverless_bundle';
import initServerlessSurveyBanner from '~/serverless/survey_banner';

document.addEventListener('DOMContentLoaded', () => {
  initServerlessSurveyBanner();
  new ServerlessBundle(); // eslint-disable-line no-new
});
