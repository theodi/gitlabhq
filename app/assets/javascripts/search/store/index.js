import Vue from 'vue';
import Vuex from 'vuex';
import createState from './state';

Vue.use(Vuex);

export const getStoreConfig = ({ query }) => ({
  state: createState({ query }),
});

const createStore = config => new Vuex.Store(getStoreConfig(config));
export default createStore;
