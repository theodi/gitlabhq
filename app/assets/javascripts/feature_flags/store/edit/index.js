import Vuex from 'vuex';
import state from './state';
import * as actions from './actions';
import mutations from './mutations';

export default data =>
  new Vuex.Store({
    actions,
    mutations,
    state: state(data),
  });
