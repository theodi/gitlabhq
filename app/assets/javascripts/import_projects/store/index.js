import Vue from 'vue';
import Vuex from 'vuex';
import state from './state';
import actionsFactory from './actions';
import * as getters from './getters';
import mutations from './mutations';

Vue.use(Vuex);

export default ({ initialState, endpoints, hasPagination }) =>
  new Vuex.Store({
    state: { ...state(), ...initialState },
    actions: actionsFactory({ endpoints, hasPagination }),
    mutations,
    getters,
  });
