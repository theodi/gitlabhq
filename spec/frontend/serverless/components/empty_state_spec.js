import { GlEmptyState, GlSprintf } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import { createStore } from '~/serverless/store';
import EmptyStateComponent from '~/serverless/components/empty_state.vue';

describe('EmptyStateComponent', () => {
  let wrapper;

  beforeEach(() => {
    const store = createStore({
      clustersPath: '/clusters',
      helpPath: '/help',
      emptyImagePath: '/image.svg',
    });
    wrapper = shallowMount(EmptyStateComponent, { store, stubs: { GlEmptyState, GlSprintf } });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('should render content', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
