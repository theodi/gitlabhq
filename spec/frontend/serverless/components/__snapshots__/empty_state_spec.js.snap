// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyStateComponent should render content 1`] = `
"<section class=\\"row empty-state text-center\\">
  <div class=\\"col-12\\">
    <div class=\\"svg-250 svg-content\\"><img src=\\"/image.svg\\" alt=\\"Getting started with serverless\\" class=\\"gl-max-w-full\\"></div>
  </div>
  <div class=\\"col-12\\">
    <div class=\\"text-content gl-mx-auto gl-my-0 gl-p-5\\">
      <h1 class=\\"h4\\">Getting started with serverless</h1>
      <p>In order to start using functions as a service, you must first install Knative on your Kubernetes cluster. <gl-link-stub href=\\"/help\\">More information</gl-link-stub>
      </p>
      <div>
        <gl-button-stub category=\\"primary\\" variant=\\"success\\" size=\\"medium\\" icon=\\"\\" buttontextclasses=\\"\\" href=\\"/clusters\\">Install Knative</gl-button-stub>
        <!---->
      </div>
    </div>
  </div>
</section>"
`;
